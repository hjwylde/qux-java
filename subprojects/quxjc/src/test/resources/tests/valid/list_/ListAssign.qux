package valid.list_

[int] f([int] xs):
    xs[0] = 3

    return xs

void main([str] args):
    xs = [0, 1, 2]
    print xs[0]
    print xs[1]
    print xs[2]
    print f(xs)[0]
    print xs[0]
    xs[1] = {1, 2}
    print xs
    xs[0] *= 3
    print xs
    xs[0] += 3
    print xs
    xs = [[[0, 1], [2, 3]]]
    print xs
    xs[0][1][1] //= 2
    print xs
    xs[0][1] = {x: [0, 1]}
    xs[0][1].x[1] = 5
    print xs

