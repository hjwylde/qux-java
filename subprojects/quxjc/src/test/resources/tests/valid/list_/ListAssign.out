0
1
2
3
0
[0, 3, 2]
[0, 3, 5]
[0, {1, 2}, 5]
[[[0, 1], [2, 3]]]
[[[0, 1], [2, 1]]]
[[[0, 1], {x: [0, 1]}]]
[1, 1, 2]
[foo, 1, 2]
[[0, 1, 2]]
[[5, 1, 2]]
[[0, 1, 2]]
